src/pcre2.h.in

```diff
@@ -1003,6 +1003,29 @@ PCRE2_SUFFIX a no-op. Otherwise, generate an error. */
 #endif
 #endif  /* PCRE2_CODE_UNIT_WIDTH is defined */
 
+/* Assertion macros */
+
+#ifdef HAVE_BUILTIN_UNREACHABLE
+#define PCRE2_UNREACHABLE() __builtin_unreachable()
+#else
+#define PCRE2_UNREACHABLE() do {} while(0)
+#endif
+
+#ifdef PCRE2_DEBUG
+#if defined(HAVE_BUILTIN_EXPECT) && defined(HAVE_BUILTIN_UNREACHABLE)
+#define PCRE2_ASSERT(x) do { if (__builtin_expect(!(x), 0)) __builtin_unreachable(); } while (0)
+#elif defined(HAVE_ASSERT_H)
+#include <assert.h>
+#define PCRE2_ASSERT(x) assert(x)
+#elif defined(HAVE_STDLIB_H) && defined(HAVE_STDIO_H)
```

Comment:
where did those HAVE_STDLIB_H and HAVE_STDIO_H variables came from? was this code copied from somewhere else where they were relevant or added intentionally?

FWIW we support building the library in multiple ways, and at least `cmake` wasn't really working until #485

Reply:
> where did those HAVE_STDLIB_H and HAVE_STDIO_H variables came from?

I am just trying to remember. I believe that I added them because the function prototype for `abort()` is defined in `stdlib.h` and the prototype for `fprintf` in `stdio.h`. However, if they are not needed, thanks for removing them. :+1:

> FWIW we support building the library in multiple ways, and at least `cmake` wasn't really working until #485

Thanks for mentioning that. I guess that the `cmake` build is not tested in CI. Would it be correct to say that all commits targeting `master` should always be tested locally both with a configure/make build and also with a cmake build?

Just an idea here: It would be helpful for new contributors if there was a 'checklist' of everything that should be done before opening a new PR. (i.e. run the tests with JIT enabled, run the tests on 8-bit, 16-bit, and 32-bit versions of the library, etc. etc.)

Reply:
> > where did those HAVE_STDLIB_H and HAVE_STDIO_H variables came from?
> 
> I am just trying to remember. I believe that I added them because the function prototype for `abort()` is defined in `stdlib.h`

but you know those variables had to be configured somewhere, right?, you did add the logic for doing so for HAVE_ASSERT_H and even did something similar for `cmake`, so the inconstancy was puzzling.

did you test all combinations of the code that was being introduced?, presume you at least did run `./configure`, but it is also possible to change the generated config.h, see `maint/README` for some tips.

> I guess that the cmake build is not tested in CI.

of course it does, you even reviewed a patch of mine that was contributed to improve on that as part of your own change.

the original Windows job failed, but we know already that the logs weren't reviewed for that one until a week later or so.

goes without saying that CI is there to help validating the new changes, but it is no way a substitute for your own testing.

do you know how to test a cmake build? as I suggested before, the yml files from the CI can be very instructive, you should also familiarize yourself with `NON-AUTOTOOLS-BUILD`

Reply:
> but you know those variables had to be configured somewhere, right?

Definitely, they need to be configured. If the code to check for the presence of those headers was missing, that was a mistake, and I apologize.

> do you know how to test a cmake build? as I suggested before, the yml files from the CI can be very instructive, you should also familiarize yourself with `NON-AUTOTOOLS-BUILD`

I'll read through `NON-AUTOTOOLS-BUILD` soon. Thanks.